db/view/row_locking.cc

```diff
@@ -74,49 +75,31 @@ row_locker::lock_pk(const dht::decorated_key& pk, bool exclusive, db::timeout_cl
     auto tracker = latency_stats_tracker(exclusive ? stats.exclusive_partition : stats.shared_partition);
     auto i = _two_level_locks.try_emplace(pk, this).first;
     auto f = exclusive ? i->second._partition_lock.write_lock(timeout) : i->second._partition_lock.read_lock(timeout);
+    co_await std::move(f);
     // Note: we rely on the fact that &i->first, the pointer to a key, never
     // becomes invalid (as long as the item is actually in the hash table),
     // even in the case of rehashing.
-    return f.then([this, pk = &i->first, exclusive, tracker = std::move(tracker)] () mutable {
-    // FIXME: indentation broken on purpose
     tracker.lock_acquired();
-        return lock_holder(this, pk, exclusive);
-    });
+    co_return lock_holder(this, &i->first, exclusive);
```

Comment:
Let's separate the coroutinization and fixing (and do the fixing first).

Reply:
okay, will do. but it will be uglier...